easybuild/easyblocks/o/openkim.py

```diff
@@ -0,0 +1,82 @@
+##
+# Copyright 2009-2016 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Python, implemented as an easyblock
+
+@author: Jakob Schiotz (Tech. Univ. Denmark)
+"""
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.tools.run import run_cmd
+from easybuild.tools.config import build_option
+import urllib2
+import json
+
+class OpenKIM(ConfigureMake):
+    def build_step(self, verbose=False, path=None):
+        """
+        Start the actual build
+        """
+        # Build the API.
+        self.log.info("Building the OpenKIM API")
+        out1 = ConfigureMake.build_step(self, verbose=verbose, path=path)
+        
+        # Download all models
+        self.log.info("Downloading all OpenKIM models")
+        out2 = self.download_kim_models(verbose, path)
```

Comment:
I wish we could.  The OpenKIM project does not make download of all models available in a single tar file; instead you can request a list of model names from their database, and then get each model one at a time as a tar file.  Fortunately, their makefile handles this (including dependencies between "Models" and "Model Drivers").

Thank you for taking the time to review this.  I will fix the things in the other comments.